content/docs/quickstart/cpp.md

```diff
@@ -3,69 +3,163 @@ layout: quickstart
-Buffers v3 for service definitions and data serialization, and our example code
-uses Protocol Buffers as well as gRPC. If you don't already have it installed on
-your system, you can install the version cloned alongside gRPC. First ensure
-that you are running these commands in the gRPC tree you just built in the from
-the previous step.
+Add the local `bin` folder to your path variable, for example:
 
 ```sh
-$ cd third_party/protobuf
-$ make && sudo make install
+$ export PATH="$PATH:$MY_INSTALL_DIR/bin"
 ```
 
-###  Build the example
+### Prerequisites
 
-Always assuming you have gRPC properly installed, go into the example's
-directory:
+#### cmake
+
+Version 3.13 or later of `cmake` is required to install gRPC locally.
+
+- Linux
+
+  ```sh
+  $ sudo apt install -y cmake
+  ```
+
+- macOS:
+
+  ```sh
+  $ brew install cmake
+  ```
+
+- For general `cmake` installation instructions, see [Installing CMake][].
+
+Check the version of `cmake`:
 
 ```sh
-$ cd examples/cpp/helloworld/
+$ cmake --version
 ```
 
-Let's build the example client and server:
+Under Linux, the version of the system-wide `cmake` can be too low. You can
+install a more recent version into your local installation directory as follows:
+
 ```sh
-$ make
+$ wget -q -O cmake-linux.sh https://github.com/Kitware/CMake/releases/download/v{{< param cmake-version >}}/cmake-{{< param cmake-version >}}-Linux-x86_64.sh
+$ sh cmake-linux.sh -- --skip-license --prefix=$MY_INSTALL_DIR
+$ rm cmake-linux.sh
 ```
 
-Most failures at this point are a result of a faulty installation (or having
-installed gRPC to a non-standard location. Check out [the installation
-instructions for details](https://github.com/grpc/grpc/blob/master/src/cpp/README.md#make)).
+
+#### gRPC and Protocol Buffers v3
+
+While not mandatory, gRPC applications usually leverage [Protocol Buffers
+v3][pbv3] for service definitions and data serialization, and the example code
+uses Protocol Buffers.
+
+The following instructions will locally install gRPC and Protocol Buffers.
+
+ 1. Install the basic tools required to build gRPC:
+
+    - Linux
+
+      ```sh
+      $ sudo apt install -y build-essential autoconf libtool pkg-config
+      ```
+
+    - macOS:
+
+      ```sh
+      $ brew install autoconf automake libtool pkg-config
+      ```
+
+ 2. Clone the `grpc` repo and its submodules:
+
+    ```sh
+    $ git clone --recurse-submodules -b {{< param grpc_release_tag >}} https://github.com/grpc/grpc
+    $ cd grpc
+    ```
+
+ 3. Build and locally install gRPC and all requisite tools:
+
+    ```sh
+    $ mkdir -p cmake/build
+    $ pushd cmake/build
+    $ cmake -DgRPC_INSTALL=ON \
+          -DgRPC_BUILD_TESTS=OFF \
+          -DCMAKE_INSTALL_PREFIX=$MY_INSTALL_DIR \
+          ../..
+    $ make -j
+    $ make install
+    $ popd
+    ```
+
+You can find a complete set of instructions for building gRPC C++ in [Building
+from source][from-source].
```

Comment:
note that https://github.com/grpc/grpc/blob/master/src/cpp/README.md provides the instructions for building from the user's perspective (and BUILDING.md is mostly meant for power users and contributors, as noted in its first paragraph).

Reply:
The README provides instructions on how to _use_ gRPC. In the quick start we are instructing readers to locally _build_ gRPC, so I'd rather keep the link to BUILDING page. WDYT?

Reply:
sg, I'll leave up to you. Just wanted to point out the dichotomy.

Reply:
On the other hand, we should at least provide a link to README.md because it describes other modes of installation with cmake (fetchcontent, as submodule, etc).

Reply:
Done